exchanges/graphcache/src/cacheExchange.ts

```diff
@@ -342,15 +342,15 @@ export const cacheExchange = (opts?: CacheExchangeOpts): Exchange => ({
     // Forward operations that aren't cacheable and rebound operations
     // Also update the cache with any network results
     const result$ = pipe(
-      forward(
-        merge([
-          pipe(
-            inputOps$,
-            filter(op => !isCacheableQuery(op))
-          ),
-          cacheOps$,
-        ])
-      ),
+      merge([
+        pipe(
+          inputOps$,
+          filter(op => !isCacheableQuery(op))
+        ),
+        cacheOps$,
+      ]),
+      filter(op => !(op.operationName === 'query' && op.context.requestPolicy === 'cache-only')),
```

Comment:
Is it fair to assume we'll never get a non-query operation with a cachePolicy?

```suggestion
      filter(op => op.context.requestPolicy !== 'cache-only'),
```